cmd/strelaysrv/listener.go

```diff
@@ -151,7 +151,11 @@ func protocolConnectionHandler(tcpConn net.Conn, config *tls.Config) {
 			case protocol.ConnectRequest:
 				requestedPeer, err := syncthingprotocol.DeviceIDFromBytes(msg.ID)
 				if err != nil {
-					panic(err)
+					if debug {
+						log.Println(id, "is looking for an invalid peer ID")
+					}
+					conn.Close()
```

Comment:
add a `protocol.WriteMessage(conn, protocol.ResponseNotFound)` maybe, to be nicer to the caller.

Reply:
Done. 8b3270bd006c780354acddd65667edcac8c4589a